.circleci/config.yml

```diff
@@ -62,20 +61,29 @@ jobs:
             echo 'export NVM_DIR="$HOME/.nvm"' >> $BASH_ENV
             echo '[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"' >> $BASH_ENV
 
-      # Build frontend JS
-      - run: cd client && yarn install && yarn build:production && cd -
+      # Restore cached Node dependencies
+      - restore_cache:
+          keys:
+            - v14.16.1-node-{{ checksum "client/yarn.lock" }}
+            # Fallback to using the latest cache if no exact match is found
+            - v14.16.1-node-
 
+      # Update Node dependencies
+      - run: cd client && yarn install && cd -
+
+      # Recache the updated Ruby dependencies
```

Comment:
Looks very clean! Thanks!

Just a minor change here "Node" and perhaps you wanna squash all the commits?